OpenDreamRuntime/Procs/Native/DreamProcNativeRoot.cs

```diff
@@ -866,6 +866,35 @@ public static DreamValue NativeProc_list2params(DreamObject instance, DreamObjec
             return currentMin;
         }
 
+        [DreamProc("nonspantext")]
+        [DreamProcParameter("Haystack", Type = DreamValueType.String)]
+        [DreamProcParameter("Needles", Type = DreamValueType.String)]
+        [DreamProcParameter("Start", Type = DreamValueType.Float, DefaultValue = 1)]
+        public static DreamValue NativeProc_nonspantext(DreamObject instance, DreamObject usr, DreamProcArguments arguments)
+        {
+            string text = arguments.GetArgument(0, "Haystack").GetValueAsString();
+            string needles = arguments.GetArgument(1, "Needles").GetValueAsString();
+            float start = arguments.GetArgument(2, "Start").GetValueAsFloat() - 1; // Decrease, because in DM the enumeration starts from 1
+
+            int index = 0;
+            if (start < 0)
+            {
+                start = text.Length + start;
+                index = text.IndexOfAny(needles.ToCharArray(), (int)start);
```

Comment:
```suggestion
                index = text.AsSpan().IndexOfAny(needles, (int)start);
```
You can do this to avoid turning the needles string into an array, which is more optimal (also do this below).

Reply:
Also, you could move this `index = text.IndexOfAny(needles.ToCharArray(), (int)start);` out of the if/else because it is identical between the two blocks. Just move it after the rest so the only thing the `if` will contain is `start = text.Length + start;`

Reply:
AsSpan().IndexOfAny() not support "start index"
![image](https://user-images.githubusercontent.com/11507166/141275254-4a9512e7-3c6d-4513-96cf-1aa58d85fc6a.png)

Reply:
fix

Reply:
Use `AsSpan((int) start).IndexOfAny(needles)` instead then. That *should* work. But note that your returned index will be relative to the start position.